mlx_engine/model_kit/model_kit.py

```diff
@@ -158,84 +129,85 @@ def tokenize(self, prompt: str) -> List[int]:
     def process_prompt(
         self,
         prompt_tokens,
-        img_b64,
+        images_b64: Optional[List[str]],
         prompt_progress_callback,
         generate_args,
         speculative_decoding_toggle: Optional[bool] = None,
-    ) -> mx.array:
-        """
-        This method processes the prompt, adding its tokens to the cache history
+    ) -> Tuple[mx.array, Optional[mx.array]]:
+        ### TEXT-ONLY PROCESS_PROMPT ###
+        is_text_only_processing = images_b64 is None or len(images_b64) == 0
+        if is_text_only_processing:
+            return process_prompt_text_only(
+                prompt_tokens,
+                self.cache_wrapper,
+                generate_args,
+                self.draft_model,
+                speculative_decoding_toggle,
+                prompt_progress_callback,
+            ), None
+        ### WITH IMAGES PROMPT PROCESSING ###s
+        if self.vision_add_on is None:
+            raise ValueError(
+                "Vision add-on is not loaded, but images were provided for processing"
+            )
+        embeddings = self.vision_add_on.compute_embeddings(
+            self.model, prompt_tokens, images_b64
+        )
+        return [], embeddings
```

Comment:
```suggestion
        return mx.array([]), embeddings
```
to match the type-hint. Or, could rethink the type-hint